### PR TITLE
🐄Update Frontend Appointments Flow

### LINKED ISSUE #268: Update Appointment Booking
#### Description
With mentee accounts, we no longer need to store information like name and languages spoken in the appointments schema as it is already stored in the account.
We need to come up with a new appointments schema while also adjusting our endpoints to ensure they work correctly. 
This is also related to #249 
- [x] Come up with new schema: feel free to get the rest of the devs involved in this discussion
- [x] update all appointments related endpoints
- [x] shorten appointment form to this 
image

- [x] test functionality: appointment data download, booking an appointmnet, mentors viewing their appointments, admins viewing all appointments

### PR DESCRIPTION
**Pull request by Leundai**
## Description :star2:

Fixes #268 
- Adjusted backend endpoints to support the new menteeID field
- made older fields legacy
- kept name within official model of AppointmentRequest to avoid requesting all Mentee's Infos
- Updated appointment view modal
- Adjusted date formatting
- Changed Appointment Creation Modal to only 4 fields, allowCalls, allowTexts, Topic, Message

## Screenshots
image

image
image

## TODOs :star:
- [X] Tests
- [X] Documentation

### CODE REVIEW

**Review by AngelaLuo49021 — CHANGES_REQUESTED** [review]
image
This top part look a little whack

File: backend/api/views/appointment.py [hunk 1]
```diff
@@ -158,15 +162,16 @@ def put_appointment(id):
...
-    res_email = send_email(
-        recipient=appointment.email,
-        subject="Mentee Appointment Notification",
-        data={"name": mentor.name, "date": start_time, "approved": True},
-        template_id=MENTEE_APPT_TEMPLATE,
-    )
-    if not res_email:
-        logger.info("Failed to send email")
+    if mentee.email_notifications:
+        start_time = appointment.timeslot.start_time.strftime(APPT_TIME_FORMAT + " GMT")
```
Comment (kenetec) [inline backend/api/views/appointment.py:166]: Try using `f'{APPT_TIME_FORMAT} GMT'` instead

File: frontend/src/components/MenteeAppointmentModal.js [hunk 2]
```diff
@@ -161,6 +136,7 @@ function MenteeAppointmentModal(props) {
       start_time: moment(time.start_time.$date).format(),
       end_time: moment(time.end_time.$date).format(),
     };
+    console.log(appointment);
```
Comment (kenetec) [inline frontend/src/components/MenteeAppointmentModal.js:139]: Remove `console.log`

**Review by kenetec — APPROVED** [review]
sweeeet 🥧

**Review by AngelaLuo49021 — APPROVED** [review]
Looks great!

**Pull request merged**